### PR TITLE
[Merged by Bors] - Add otp.md and ibc.md

### PR DESCRIPTION
**Pull request by KaiserKarel**
## Issue
*Please replace this line with either a link to the ClickUp issue, or a reference to the GitHub issue.*

## Description
*Please replace this line with info that is not in the ClickUp ticket, eg: "also bumps dependency foo version"*

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: xcvm/architecture/transports/otp.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# Open Transport Protocol
```
Comment (hussein-aitlahcen) [inline xcvm/architecture/transports/otp.md:1]: Add (OTP) at the end to make the acronym definition explicit

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

File: xcvm/architecture/transports/otp.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    Governance->>Gateway: Settle dispute.
+```
+
+Validators can be any entity, including a user triggering the dispute through our UI, as well as nodes automatically tracking registrations. For bridging teams, it is especially important to dispute registrations of your bridge with an incorrect administrator account.
+
+## Egress of messages
+
+Once a transport has been succesfully registered, it becomes accessible to the `Gateway`, which might route traffic based on transport requirements from the interpreter instance dispatching the program. By default, all OTP transports are considered `Trusted`, meaning that no traffic is routed through them unless specifically opted-in by the interpreter instance. 
+
+Later we may add means through governance to upgrade transports to the `Trustless` status using governance. This will accomodate transports such as `XCMP`.
```
Comment (benluelo) [inline xcvm/architecture/transports/otp.md:78]:
```suggestion
Later we may add means to upgrade transports to the `Trustless` status using governance. This will accomodate transports such as `XCMP`.
```

File: xcvm/architecture/transports/otp.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+A note on this, fees are specified by the program, meaning that the bridge itself cannot charge a fee directly on the `Assets` being transfered. The third-party OTP-adapter will need to take this into account, by possibly using internal APIs of the bridging contract to ensure that no fees are charged based on the `Assets` transferred.
+
+## Reimbursements
+
+When a transports fails to transmit a message (such as an IBC timeout or network congestion) the `Third-party Transport` issues a reimbursement to the `Gateway` which then informs the interpreter instance of the failure.
+
+For `OTP`-registerd transports, the reimbursement itself can be disputed as well. After issueing the reimbursement to the Gateway, a short-lived dispute window (hours) allows validators to dispute the reimbursement. This avoids the attack vector where a bridge has identified a contract which will retry a cross-chain operation until success, and extract value from that by issueing malicious reimbursements (a cross-chain re-entrancy attack).
```
Comment (benluelo) [inline xcvm/architecture/transports/otp.md:97]:
```suggestion
For `OTP`-registered transports, the reimbursement itself can be disputed as well. After issuing the reimbursement to the Gateway, a short-lived dispute window (hours) allows validators to dispute the reimbursement. This avoids the attack vector where a bridge has identified a contract which will retry a cross-chain operation until success, and extract value from that by issuing malicious reimbursements (a cross-chain re-entrancy attack).
```

File: xcvm/architecture/transports/otp.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+## XCVM Ingress
+
+A transport does not need to register itself to pass messages into the `XCVM`, it can do so directly, assuming that the interpreter instance accepting the transport's messages has whitelisted this transport. 
+
+The following diagram describes the flow of data (and funds) when OTP is used to for ingress of programs.
+
+```mermaid
+sequenceDiagram
+    Third-party transport->>Third-party OTP-adapter: Transform messsage.
```
Comment (dzmitry-lahoda) [inline xcvm/architecture/transports/otp.md:33]: message

File: xcvm/architecture/transports/otp.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+## Blacklist
+
+The blacklist is used to enact emergency operations in case of bridge fraud, which can be caused by hacks, or bugs within the system. The blacklist globally disables bridges based on `LAYR` governance.
+
+Note that the blacklist is not used to manage ecosystem specific risk, such as toxic assets. It only concerns itself with bridges where the message authenticity can no longer be guaranteed.
+
+## XCVM Ingress
+
+A transport does not need to register itself to pass messages into the `XCVM`, it can do so directly, assuming that the interpreter instance accepting the transport's messages has whitelisted this transport.
```
Comment (dzmitry-lahoda) [inline xcvm/architecture/transports/otp.md:27]: not clear what it means. is that chain which is initiator of XCVM, creates XCVM instance with whilelisted set of transports?  and then that list transferred to XCVM on receiving side chain?

File: xcvm/architecture/transports/otp.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+A transport does not need to register itself to pass messages into the `XCVM`, it can do so directly, assuming that the interpreter instance accepting the transport's messages has whitelisted this transport. 
+
+The following diagram describes the flow of data (and funds) when OTP is used to for ingress of programs.
+
+```mermaid
+sequenceDiagram
+    Third-party transport->>Third-party OTP-adapter: Transform messsage.
+    Third-party OTP-adapter->>Gateway: Transfer assets and pass program.
+    Gateway->>Router: Add bridging info and transfer funds.
+    Note over Gateway: Accept/deny program based on blacklist.
```
Comment (dzmitry-lahoda) [inline xcvm/architecture/transports/otp.md:36]: is Gateway singleton per chain which is governed by LAYR?

File: xcvm/architecture/transports/otp.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+Immediately after registration, the bridge registration can be disputed for approximately 72 hours. The stake can be lost during this time, if governance determines that the registration is fraudulent. If a dispute is triggered, the registration can be delayed by up to 2 weeks.
+
+#### Disputed Registration
+
+The following diagram displays the sequence of a registration, including the dispute by a validator.
+
+```mermaid
+sequenceDiagram
+    Third-party Transport->>Gateway: Register bridge for egress.
```
Comment (dzmitry-lahoda) [inline xcvm/architecture/transports/otp.md:65]: is Gateway will be only one chain as some root (Composable/Picasso), and stuff copied to other chains? I am thinkin about if XCVM will be initiated on other chains?

File: xcvm/architecture/transports/otp.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+For outgoing messages, the pattern is very similar:
+
+```mermaid
+sequenceDiagram
+    Gateway->>Adapter: Submit Spawn.
+    Note over Gateway,Adapter: Selects based on destinations.
+    Adapter->>Bridge Contract: Call transfer.
+    Bridge Contract->>Multisig Relayer: Enact token transfer.
+    Adapter->>Bridge Contract: Call message pass.
+    Bridge Contract->>Multisig Relayer: Enact token transfer.
```
Comment (dzmitry-lahoda) [inline xcvm/architecture/transports/otp.md:168]: Enact token transfer. > Enact message pass

**Review by dzmitry-lahoda — COMMENTED** [review]
could be helpful on diagram:

- use some indicator components - is only what has `Contract` is on chain? All which is off chain 
- has not `contract` word?  I used boxes for on chain component. So all on chain in box of chain 1 and other in box of chain too. Box are named so can get name XCVM origin(egress?) chain and egress chain? for me it hard to follow egress and ingress so
- color components which are built by composable and part of xcvm code base, parts which are fully 3rd party and not aware of XCVM and parts which are itegrations.

File: xcvm/architecture/transports/otp.md (hunk 6 above)
Comment (dzmitry-lahoda) [reply xcvm/architecture/transports/otp.md:36]: so may be clarify what is on chain vs not in this diagram?

Comment (KaiserKarel) [reply xcvm/architecture/transports/otp.md:36]: Yeah, singleton (contract) per chain

File: xcvm/architecture/transports/otp.md (hunk 7 above)
Comment (KaiserKarel) [reply xcvm/architecture/transports/otp.md:65]: No, we regularly deploy the contracts; and set the admin to Identity: (NetworkId::Composable, SomeContractAddress), which can then use IBC to configure the gateway further. `SomeContractAddress` can actually be our democracy pallet for example

**Comment by KaiserKarel** [comment]
> could be helpful on diagram:
> 
> * use some indicator components - is only what has `Contract` is on chain? All which is off chain
I'll have a look; mermaid.js is a bit limited.

> * has not `contract` word?  I used boxes for on chain component. So all on chain in box of chain 1 and other in box of chain too. Box are named so can get name XCVM origin(egress?) chain and egress chain? for me it hard to follow egress and ingress so
I wanted to keep this abstract enough to allow for pallets and modules to also provide this implementation. I can make the language more concrete though.

> * color components which are built by composable and part of xcvm code base, parts which are fully 3rd party and not aware of XCVM and parts which are itegrations.
Agree, will see if I can colour that, and otherwise put it in writing.

File: xcvm/architecture/transports/otp.md (hunk 5 above)
Comment (KaiserKarel) [reply xcvm/architecture/transports/otp.md:27]: The receiving side needs to have it whitelisted first, so if a contract is automatically setting up instances, it would do so in the following way:

1. Create instructions and send to other chain over IBC
2. On other chain, execute instructions. One of these instructions should contain Call(XcvmInterpreter::SetFilter(address)), where address is a contract that has signature: fn(bridge, identity, program) -> result<bool, str>. If it is unset, the xcvm interpreter will check that bridge == ibc. If set to something custom, the 3rd party dev can do it based on the 3 inputs, e.g. accept programs over wormhole if the program does not contain Transfer.

**Comment by KaiserKarel** [comment]
So colouring the sequence diagrams is blocked by: https://github.com/mermaid-js/mermaid/issues/523

**Review by aeryz — APPROVED** [review]

File: xcvm/architecture/transports/otp.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    Validator->>Gateway: Dispute the registration.
+    Note over Validator, Gateway: Validator puts down a stake as well.
+    Governance->>Gateway: Settle dispute.
+```
+
+Validators can be any entity, including a user triggering the dispute through our UI, as well as nodes automatically tracking registrations. For bridging teams, it is especially important to dispute registrations of your bridge with an incorrect administrator account.
+
+## Egress of messages
+
+Once a transport has been successfully registered, it becomes accessible to the `Gateway`, which might route traffic based on transport requirements from the interpreter instance dispatching the program. By default, all OTP transports are considered `Trusted`, meaning that no traffic is routed through them unless specifically opted-in by the interpreter instance.
```
[the remark this answers is not in the document]
Comment (hussein-aitlahcen) [reply xcvm/architecture/transports/otp.md:78]: Once a transport has been registered, the selection is solemnly based on the xcvm program requirement. For instance, deterministic relayers should be interchangeable

[the remark this answers is not in the document]
Comment (KaiserKarel) [reply xcvm/architecture/transports/otp.md:78]: Not specified yet; although it will be some form of:

1. Prefer IBC
2. Then XCM
3. Then OTP, based on performance, fee and stake(?).

**Review by dzmitry-lahoda — APPROVED** [review]
thanks for clarification

**Comment by aadamsx** [comment]
Did this PR add styles to Sequence Diagrams?
